GVFS/GVFS/CommandLine/MaintenanceVerb.cs

```diff
@@ -0,0 +1,104 @@
+﻿using CommandLine;
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Maintenance;
+using GVFS.Common.Tracing;
+
+namespace GVFS.CommandLine
+{
+    [Verb(MaintenanceVerbName, HelpText = "Runs a maintance job.")]
+    public class MaintenanceVerb : GVFSVerb.ForExistingEnlistment

```

Comment:
What's the reason for adding a maintenance verb?  

`LooseObjectsStep` is scheduled to run automatically, when would (or how would) a user know that they should run the maintenance verb?

Reply:
The reason is for:

1. Functional tests. We'll run the step explicitly instead of relying on background timing.
2. Support. When we are supporting someone with a bad data shape, we can ask them to run the verb.

It's also possible that we will want to add notifications like "You just hydrated your repo, consider dehydrating and cleaning up loose objects by running..."

Reply:
@wilbaker how would a user know to run dehydrate?

Reply:
Will there be any issues if a user discovers or we tell them about this verb and they decide to schedule it to run on their own at a low interval say every minute?

Reply:
@kewillford no more so if the user sets up 'git prune-packed' to run every minute ;-)

Their mount wouldn't fall over, but I wouldn't recommend doing that.  

We have telemetry on MaintenanceVerb so we could detect if someone does that.

Have users done similar things in the past or suggestions to steer away from that?

Reply:
> Will there be any issues if a user discovers or we tell them about this verb and they decide to schedule it to run on their own at a low interval say every minute?

I think my concern is related to Kevin's comment.  Are the benefits of the new verb worth expanding out CLI's API surface and supporting it going forward?

For the specific scenarios mentioned by Stolee:

> Functional tests. We'll run the step explicitly instead of relying on background timing.

Could the functional tests just call `prune-packed` rather than relying on the verb?

> Support. When we are supporting someone with a bad data shape, we can ask them to run the verb.

Could we also just as the user to run `prune-packed` manually?  I think one difference between this support task and dehydrating is that there aren't manual steps for dehydrating as it requires calls to the ProjFS API. 

> @wilbaker how would a user know to run dehydrate?

When the dehydrate feature was added it was communicated to the organization using VFS4G via email and documented in their internal wiki.

You bring up a good point though, for new users to VFS4G dehydrate is not easily found. However, I'm still not sure we need to add a new verb if we can provide a git command that can be run manually to achieve the same result.  There is no manual alternative to `gvfs dehydrate`.

Reply:
> Functional tests. We'll run the step explicitly instead of relying on background timing.

Sorry I think I misinterpreted this.  I was thinking we'd want to call it to clean stuff up, and you might have meant to test out the functionality itself.